action-server/alembic/versions/2c2215d3e953_initial_db_setup.py

```diff
@@ -0,0 +1,124 @@
+"""Initial db setup.
+
+Revision ID: 2c2215d3e953
+Revises:
+Create Date: 2020-04-21 13:32:44.422928
+
+"""
+import structlog
+from sqlalchemy import Boolean, Column, Date, DateTime, String, Time, func
+from sqlalchemy.dialects.postgresql import ENUM, UUID
+from sqlalchemy.sql.schema import ForeignKey
+
+from alembic import op
+
+log = structlog.get_logger(__name__)
+
+revision = "2c2215d3e953"
+down_revision = None
+branch_labels = None
+depends_on = None
+
+# create_type=False for ENUM: see
+# https://bitbucket.org/zzzeek/alembic/issues/53/create-table-dont-create-enum
+
+canadian_provinces = ENUM(
+    "AB",
+    "BC",
+    "MB",
+    "NB",
+    "NL",
+    "NS",
+    "NT",
+    "NU",
+    "ON",
+    "PE",
+    "QC",
+    "SK",
+    "YT",
+    name="canadian_province",
+    create_type=False,
+)
+
+languages = ENUM("en", "fr", name="language", create_type=False,)
+
+symptoms_severity = ENUM(
+    "none", "moderate", "severe", name="symptoms_severity", create_type=False,
+)
+
+
+def upgrade():
+    canadian_provinces.create(bind=op.get_bind(), checkfirst=True)
+    languages.create(bind=op.get_bind(), checkfirst=True)
+    symptoms_severity.create(bind=op.get_bind(), checkfirst=True)
+
+    op.execute('CREATE EXTENSION IF NOT EXISTS "uuid-ossp";')
+
+    create_user_table()
+    create_assesment_table()
+
+
+def create_user_table():
+    log.info("Creating user table.")
+
+    op.create_table(
+        "user",
+        Column(
+            "id",
+            UUID(as_uuid=True),
+            unique=True,
+            server_default=func.uuid_generate_v4(),
+        ),
+        Column("first_name", String()),
+        Column("created_at", DateTime(timezone=True), nullable=False),
+        Column("last_reminded_at", DateTime(timezone=True)),
+        Column("reminder_time", Time(timezone=True), nullable=False),
+        Column("preferred_language", languages, nullable=False),
+        Column("resides_in", canadian_provinces, nullable=False),
+        Column("date_of_birth", Date(), nullable=False),
```

Comment:
Before merging, we need to check with Linda if the dialogue will be changed to ask "date/year of birth" instead of age group. If the dialogue only asks for year of birth, will we still store a Date?

Reply:
Yes but I'll have to put a constraint on it basically all date will be January 1st of the birth year, I think this is the preferred way to do it since it's easier to change later.